backend-go/lib.go

```diff
@@ -0,0 +1,324 @@
+	}
+
+	return &mConf, nil
+}
+
+// Type of config that is returned after parsing
+type TelestionBaseConfig struct {
+	Dev         bool     `json:"DEV"`
+	Nats        bool     `json:"NATS"`
+	NatsUrl     string   `json:"NATS_URL"`
+	NatsUsr     string   `json:"NATS_USER"`
+	NatsPwd     string   `json:"NATS_PASSWORD"`
+	ConfigFile  string   `json:"CONFIG_FILE"`
+	ConfigKey   string   `json:"CONFIG_KEY"`
+	ServiceName string   `json:"SERVICE_NAME"`
+	DataDir     string   `json:"DATA_DIR"`
+	NonFlagArgs []string `json:"NON_FLAG_ARGS"`
+
+	CustomConfig map[string]any `json:"-" ,mapstructure:",remain"`
+}
+
+func inferTelestionConfig(mapping map[string]any) (*TelestionBaseConfig, *TelestionError) {
+	tConfig := TelestionBaseConfig{}
+
+	decoderConfig := &mapstructure.DecoderConfig{
+		ErrorUnused:      false,
+		ErrorUnset:       false,
+		WeaklyTypedInput: true,
+		Result:           tConfig,
+	}
+
+	tDecoder, err := mapstructure.NewDecoder(decoderConfig)
+	if err != nil {
+		return nil, NewTelestionError(err, "Decoder for TelestionBaseConfig inference could not be initialized!")
+	}
+
+	if err := tDecoder.Decode(mapping); err != nil {
+		return nil, NewTelestionError(err, "TelestionBaseConfig could not be inferred from given map!")
+	}
+
+	return &tConfig, nil
+}
+
+func StartService() (*nats.Conn, *TelestionError) {
+	// Ensure the config contains the minimum required fields
+	configMapping := loadConfig()
+	if _, err := inferMinimalConfig(configMapping); err != nil {
+		return nil, err
+	}
+
+	// Parse config into usable data format
+	cfgP, err := inferTelestionConfig(configMapping)
+	Config = *cfgP
+	if err != nil {
+		return nil, err
+	}
+
+	// Init NATS
+	if Config.Nats {
+		return initializeNats()
+	} else {
+		return nil, NewTelestionError(nil,
+			"Initializing a telestion service without NATS is currently not possible! "+
+				"To continue set the corresponding NATS flag to true again.")
+	}
+}
+
+func loadConfig() map[string]any {
+	config := map[string]any{
+		"CONFIG_FILE": nil,
+		"CONFIG_KEY":  nil,
+	}
+	updateConfig(config, parseConsoleArguments())
+	populateConfig(config, getEnvs())
+	if _, contained := config["DEV"]; !contained {
+		config["DEV"] = false
+	}
+
+	populateConfig(config, getDefaultConfig(config["DEV"].(bool)))
+	populateConfig(config, getFileConfig(config))
+	return config
+}
+
+func updateConfig[V any | string](config map[string]any, updates map[string]V) {
+	for k, v := range updates {
+		config[k] = v
+	}
+}
+
+func populateConfig[V any | string](config map[string]any, updates map[string]V) {
+	for k, v := range updates {
+		// check if key is already in map and populate if it isn't
+		if _, contained := config[k]; !contained {
+			config[k] = v
+		}
+	}
+}
+
+func getFileConfig(config map[string]any) map[string]any {
+	// Note that the file config is supposed to be a json config!
+	jsonConfig := map[string]any{}
+
+	configPath := config["CONFIG_FILE"].(string)
```

Comment:
This didn't work in my testing. I got this error when I didn't have a config file (which shouldn't be a requirement)

```text
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
main.getFileConfig(0x140001191a0)
        /Users/zklaschka/WebstormProjects/telestion/backend-go/lib.go:161 +0x260
main.loadConfig()
        /Users/zklaschka/WebstormProjects/telestion/backend-go/lib.go:138 +0x3dc
main.StartService()
        /Users/zklaschka/WebstormProjects/telestion/backend-go/lib.go:104 +0x1c
main.main()
        /Users/zklaschka/WebstormProjects/telestion/backend-go/sample1.go:17 +0x28
exit status 2
```

when running the service

Reply:
Fixed